src/csc.rs

```diff
@@ -134,6 +134,24 @@ impl<'a> CscMatrix<'a> {
             });
         }
     }
+
+    /// Constructs a matrix from an iterator
+    ///
+    /// The iterator should deliver values in column major format of a dense data block, otherwise 
+    /// there are more efficient methods to construct the sparse matrix. Note that this copies all
+    /// elements of the matrix.
+    ///
+    pub fn from_iterator_dense<'b, I: IntoIterator<Item=&'b float>>(nrows: usize, ncols: usize, iter: I) -> CscMatrix<'static> {
```

Comment:
Could you change this to take `Item=float`. It's just slightly more idiomatic.

Reply:
Ok, I modified this. The function will now only accept owned data.